### PR TITLE
(PA-1918) Remove mcollective

### PR DESCRIPTION
**Pull request by ekinanp**
We will no longer be shipping mcollective in puppet-agent 6.

The main reason for the deprecation is that many of mcollective's use
cases (triggering puppet runs, running remote tasks) are captured by
Orchestrator and Puppet Tasks. Thus, we are removing it from the
puppet-agent package so that it can be released separately from the
Puppet Platform.

### CODE REVIEW

**Comment by ekinanp** [comment]
I did an ad-hoc build of the changes in this branch. Here is the integration test step:

https://jenkins-master-prod-1.delivery.puppetlabs.net/job/experimental_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/3/

^ The puppet test failures are not related to the changes here, because at the time of writing this comment they also failed in our public master pipelines for the same reason (here: https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/Public%20Suites/view/master/view/Suite/job/platform_puppet-agent_puppet-agent-integration-suite_daily-master/141/).

I tried to rekick the adhoc-pipeline without the puppet tests and just run PEZ, but later found that I needed a few extra build targets for PEZ to work.

So what I did was re-kick my entire adhoc pipeline to include the PEZ build targets, and skip the Puppet component tests. Here is the result of the integration test step:

https://jenkins-master-prod-1.delivery.puppetlabs.net/job/experimental_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/5/

And here is the result of the PEZ step:

https://jenkins-master-prod-1.delivery.puppetlabs.net/job/experimental_puppet-agent-extra_puppet-agent-modules-and-pez_adhoc-ad_hoc/2/

So it seems like we can rip out mcollective from the agent.

I can also re-kick my adhoc pipeline once the Puppet tests are fixed for the master branch if we really want to be sure that things pass before merging this.

**Comment by ekinanp** [comment]
We should also double check https://github.com/puppetlabs/ci-job-configs/pull/4281 before merging this.

**Comment by ekinanp** [comment]
@Sharpie thanks for the review! do you have suggestions for how I should phrase the commit message?

**Comment by Sharpie** [comment]
The bit about MCollective's use cases being taken over by Orchestrator and Puppet Tasks is great. I'd follow that up with something like:

> MCollective is being removed from the puppet-agent package so that it can be released separately from the Puppet Platform.

**Comment by ekinanp** [comment]
@Sharpie Updated the commit message, how does it look now?

**Comment by geoffnichols** [comment]
Overall, looks good to me! 

It's a bit of an outlier, but we may also be able to remove https://github.com/puppetlabs/puppet-agent/blob/master/resources/aix/mcollective.service.

**Comment by ekinanp** [comment]
@geoffnichols Good catch! Removed it.

File: configs/projects/puppet-agent.rb [hunk 1]
```diff
@@ -89,7 +89,6 @@
   proj.component "hiera"
   proj.component "leatherman"
   proj.component "cpp-hocon"
-  proj.component "marionette-collective"
```
Comment (geoffnichols) [inline configs/projects/puppet-agent.rb]: Not able to comment on the files lines here, but it looks like there are a few mcollective-specific project settings - e.g. `mco_root`, `mco_bindir`, `mco_libdir` - that we may also be able to remove?

**Comment by ekinanp** [comment]
Actually @geoffnichols there were several more areas that needed to be modified. I'm going to re-kick the adhoc run (with the PEZ step) and see if things still pass.

**Comment by branan** [comment]
@ekinanp have you looked into the Runtime to see what we can remove from that? I believe we have at least one mcollective dependency in there (Stomp)

**Comment by branan** [comment]
We'll have to merge this before we can clean out the runtime, but I'd like to have that work staged and ready to go so we don't forget about it

**Comment by ekinanp** [comment]
@branan Yup, at the time I didn't think changes needed to be made to runtime because ruby-stomp is used by 1.10.x, 5.3.x, 5.5.x but now that you mention it, I think I was supposed to remove it as a shared agent component. https://github.com/puppetlabs/puppet-runtime/pull/45 does that.

**Comment by ekinanp** [comment]
Waiting on a passing ad-hoc run still, https://github.com/puppetlabs/puppet/pull/6793 details some comments on why. But this is blocked on https://github.com/puppetlabs/puppet/pull/6793 being merged.

Note there's an extra puppet.json commit here I will revert once ad-hoc passes -- that is just pointed to my fork of Puppet containing the above PR.

In the meantime, I decided to double check that MCO was gone on Windows, EL, AIX, OSX and Solaris. Doing a find w/ "mco", "marionette", "collective" on the filesystem on those platforms indicates that this PR would completely remove MCO, as desired.

**Comment by ekinanp** [comment]
jenkins please test this

**Comment by ekinanp** [comment]
jenkins please test this

**Comment by ekinanp** [comment]
jenkins please test this

**Comment by ekinanp** [comment]
jenkins please test this

**Comment by ekinanp** [comment]
jenkins please test this

**Comment by ekinanp** [comment]
jenkins please test this

**Comment by ekinanp** [comment]
jenkins please test this

**Comment by ekinanp** [comment]
https://jenkins-master-prod-1.delivery.puppetlabs.net/view/__experimental/job/experimental_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/ shows a passing ad-hoc run.

The puppet-agent failures are the resource api gem failures we're seeing in the master suite pipeline right now, so they are unrelated to the work done in this PR.

**Comment by ekinanp** [comment]
This is now unblocked.

**Pull request merged**